### ISSUE TITLE
Sprite.debug drawing leaks render state

### ISSUE THREAD

**Issue by islemaster** [molleindustria/p5.play#54]
The debug drawing code found in [Spirte.display() at p5.play.js:1485](https://github.com/molleindustria/p5.play/blob/master/lib/p5.play.js#L1485) does not properly push/pop p5 render state, so it can affect rendering operations that take place after it runs.

# Example
An easy example of this is that text rendering breaks when it follows sprite rendering with debug=true, including on subsequent frames.  Here's a small repro case where debug=true when the mouse is down.

![p5-debug-bug](https://cloud.githubusercontent.com/assets/1615761/13967390/1202793e-f034-11e5-8b7b-a70908e4d3e0.gif)

You can run this example at http://p5play.molleindustria.org/examples/ by pasting the following code:
```javascript
var circle;

function setup() {
  createCanvas(800,400);  
  
  //create 4 sprites
  circle = createSprite(400, 200);
  //compact way to add an image
  circle.addImage(loadImage("assets/plain_circle.png"));
  
}

function draw() {
  background(255,255,255);  
  

  //if debug is set to true bounding boxes, centers and depths are visualized
  circle.debug = mouseIsPressed;
  
  drawSprites();
  
  textAlign(CENTER, CENTER);
  text('Here is some example text', 0, 0, 800, 200);
}
```

# Related issue
There's a related problem, equally easy to fix, where debug also doesn't draw if the user calls `strokeWeight(0)`.

@joshlory has volunteered to fix these two issues; assigning to him right away.

**Comment by islemaster** [comment]
Reopening this issue until processing/p5.js#1320 is resolved.

**Comment by islemaster** [comment]
Fix is merged upstream, all that's left is to wait for p5.js to cut a new version and then update to that version in this repo.  Any idea what their process for cutting a version is?  Looks like the last one was a few weeks ago.

**Comment by toolness** [comment]
Oh, I don't actually know what the process/logic behind cutting a version is :disappointed:

**Comment by islemaster** [comment]
Fixed as of #86 hooray!  FYI @joshlory
